code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -101,16 +101,13 @@ export function composeStory<TRenderer extends Renderer = Renderer, TArgs extend
       parameters: story.parameters as Parameters,
       argTypes: story.argTypes as StrictArgTypes<TArgs>,
       id: story.id,
-      play: (async (extraContext: ComposedStoryPlayContext<TRenderer, TArgs>) => {
-        if (story.playFunction === undefined) {
-          throw new Error('The story does not have a play function. Make sure to add one.');
-        }
-
-        await story.playFunction({
-          ...context,
-          ...extraContext,
-        });
-      }) as unknown as ComposedStoryPlayFn<TRenderer, Partial<TArgs>>,
+      play: story.playFunction
+        ? ((async (extraContext: ComposedStoryPlayContext<TRenderer, TArgs>) =>
+            story.playFunction!({
+              ...context,
+              ...extraContext,
+            })) as unknown as ComposedStoryPlayFn<TRenderer, Partial<TArgs>>)
+        : undefined,
```

Comment:
nit: personally I find this syntax more readable in these cases, but it's totally up to you.

```suggestion
      ...(story.PlayFunction && { play: ((async (extraContext: ComposedStoryPlayContext<TRenderer, TArgs>) =>
            story.playFunction!({
              ...context,
              ...extraContext,
            })) as unknown as ComposedStoryPlayFn<TRenderer, Partial<TArgs>>) })
```